src/Symfony/Component/Security/Core/Authorization/AccessDecisionManager.php

```diff
@@ -89,6 +89,8 @@ private function decideAffirmative(TokenInterface $token, array $attributes, $ob
 
             if (VoterInterface::ACCESS_DENIED === $result) {
                 ++$deny;
+            } elseif (VoterInterface::ACCESS_ABSTAIN !== $result) {
+                trigger_deprecation('symfony/security-core', '5.3', 'Returning "%s" in "%s::vote()" is deprecated, return one of "%s" constant: "ACCESS_GRANTED", "ACCESS_DENIED" or "ACCESS_ABSTAIN".', get_debug_type($result), get_debug_type($voter), VoterInterface::class);
```

Comment:
```suggestion
                trigger_deprecation('symfony/security-core', '5.3', 'Returning "%s" in "%s::vote()" is deprecated, return one of "%s" constants: "ACCESS_GRANTED", "ACCESS_DENIED" or "ACCESS_ABSTAIN".', get_debug_type($result), get_debug_type($voter), VoterInterface::class);
```

Reply:
If my voter returns an arbitrary integer (`42` for instance), I will get a deprecation message that tells me that returning an int is deprecated. This is a bit misleading because the constants that we accept are actually integers.

Reply:
Good point. replaced `get_debug_type` by `var_export`.